src/backend/tests/Fusion.Resources.Api.Tests/IntegrationTests/PersonAbsenceTests.cs

```diff
@@ -98,6 +101,110 @@ public async Task DeleteAbsence_ShouldBeOk_WhenAdmin()
+                AbsencePercentage = null,
+                IsPrivate = false,
+                TaskDetails = new ApiTaskDetails
+                {
+                    BasePositionId = basePosition.Id,
+                    TaskName = "task name",
+                    RoleName = null,
+                    Location = "location"
+                }
+            };
+
+            using var authScope = fixture.AdminScope();
+            var response = await client.TestClientPutAsync<TestAbsence>($"/persons/{testUser.AzureUniqueId}/absence/{TestAbsenceId}", request);
+            response.Should().BeSuccessfull();
+
+            response.Value.TaskDetails?.RoleName.Should().Be(BasePositionName);
+        }
+
+        [Fact]
+        public async Task PrivateNotesShouldBeHiddenForOtherResourceOwners()
+        {
+            var siblingResourceOwner = fixture.AddProfile(FusionAccountType.Employee);
+            siblingResourceOwner.FullDepartment = "TPD PRD TST QWE ABC";
+            siblingResourceOwner.Department = "TST QWE ABC";
+            siblingResourceOwner.IsResourceOwner = true;
+
+            var request = new CreatePersonAbsenceRequest
+            {
+                AppliesFrom = new DateTime(2021, 04, 30),
+                AppliesTo = new DateTime(2022, 04, 30),
+                Comment = "A comment",
+                Type = ApiPersonAbsence.ApiAbsenceType.OtherTasks,
+                AbsencePercentage = null,
+                IsPrivate = true,
+
+                TaskDetails = new ApiTaskDetails
+                {
+                    TaskName = "Top secret task name",
+                    RoleName = "Top secret role name",
+                    Location = "Top secret location"
+                }
+            };
+
+            using var authScope = fixture.AdminScope();
+            var response = await client.TestClientPostAsync<TestAbsence>($"/persons/{testUser.AzureUniqueId}/absence/", request);
+            response.Should().BeSuccessfull();
+
+            using var userScope = fixture.UserScope(siblingResourceOwner);
+            response = await client.TestClientGetAsync<TestAbsence>($"/persons/{testUser.AzureUniqueId}/absence/{response.Value.Id}");
+            response.Should().BeSuccessfull();
+
+            var taskDetails = response.Value.TaskDetails;
+            taskDetails.Should().NotBeNull();
+
+            taskDetails!.IsHidden.Should().Be(true);
+
+            response.Value.Comment.Should().NotBe(request.Comment);
+            taskDetails!.TaskName.Should().NotBe(request.TaskDetails.TaskName);
+            taskDetails!.RoleName.Should().NotBe(request.TaskDetails.RoleName);
+            taskDetails!.Location.Should().NotBe(request.TaskDetails.Location);
+        }
+
+        [Theory]
+        [InlineData(ApiPersonAbsence.ApiAbsenceType.Absence)]
+        [InlineData(ApiPersonAbsence.ApiAbsenceType.Vacation)]
+        public async Task SettingTaskDetailsOnOtherTypesShouldFail(ApiPersonAbsence.ApiAbsenceType type)
```

Comment:
Naming.. not sure what we are testing here if only reading summary of test output..